### DISCUSSION TITLE
Strange CAN frames causing trouble

### DISCUSSION THREAD

**Discussion by ARanvik** [canopen-python/canopen#289] [General]
Hello,

If doing a simple expedited SDO exchange with a node, it every other time fails with a message saying that "struct.unpack requires at least 1 byte". To debug this i used cansend to manually send the CAN frame and see what happens with candump:

can0 601 [8] 40 18 10 01 00 00 00 00 # I send request
can0 581 [0]                                        # Weird response from node
can0 601 [8] 40 18 10 01 00 00 00 00 # Send request again..
can0 581 [8] 43 18 10 01 47 01 00 00 # This is the CORRECT response from the node
can0 601 [8] 40 18 10 01 00 00 00 00 # Send request again..
can0 581 [0]                                        # Again, weird response from node

This continues in the same way, every other request responds with this weird zero payload response.

Is there any way that this is caused by a wrong setup of the socketCAN linux driver? Or is this data actually coming from the node?

**Comment by vikingsraven** [comment]
HI, does the data have 2 different transmit periods? e.g.  601 @ 50ms and 581 @ 100 ms?
im still learning this but thats what it looks like to me

**Reply by ARanvik** [reply]
Hello and thanks for your reply. I am really not sure what was wrong.. I noticed that on my MCP2515 CAN Shield from SeeedStudio, the OBD Power DIP switch was in the ON position even though it was not in use, this may have been the cause, but I have not re-tested this since then. I switched out the adapter with a Kvaser LeafLight V2 USB adapter instead. The canopen library works perfectly for me now after I installed CANLib (from Kvaser) on my Raspberry Pi 4B.

**Reply by vikingsraven** [reply]
No worries, just trying to get some help myself as well lol.
